### ISSUE TITLE
[Bug] Upon tracking a manga on MAL, Tachiyomi seems to send some information, causing the Last Updated date to be modified

### ISSUE THREAD

**Issue by paulo27ms** [tachiyomiorg/tachiyomi#4373]
## Device information
App version: 0.10.7-3063 (standard, 17109ab7, 54)
Android version: 10 (SDK 29)
Android build ID: QKQ1.190825.002 test-keys
Device brand: Xiaomi
Device manufacturer: Xiaomi
Device name: davinci
Device model: Mi 9T
Device product name: davinci_eea

## Steps to reproduce
Track a manga (remove the track and retrack to test on already tracked manga)

### Expected behavior
Nothing special should happen.

### Actual behavior
Tachiyomi seems to be sending information to MAL, causing the Last Updated date to be updated.
My guess is that it's sending the exact same data it got back to MAL. There's absolutely no reason for this to happen and it's really unwanted behavior.

It also seems like in the previous dev build (last week), when you used the refresh all tracking metadata button in the settings, it did this for every manga it refreshed, meaning a good chunk of my list got this field updated. This no longer seems to be the case as I can't reproduce it anymore so I'm guessing one of the changes related to the MAL tracker in the past week fixed this (I only noticed that this happened to my list after I updated Tachiyomi). If I get sometime this week I'll grab the last build and test to see if it's still happening in that build to confirm if it has been fixed or not or if there's other factors I'm not accounting for now.

If you're unaware what this field is, it updates every time there's something modified in the entry (chapter added, score changed, etc).
You can sort your list to use that date like this (sorting number is 5): https://myanimelist.net/mangalist/Paulo27?status=3&order=5
And you can view it by clicking the More button: 
![IMG_20210131_223052](https://user-images.githubusercontent.com/13546823/106400041-0753fb80-6414-11eb-9769-5ae54ca354d1.jpg)

**Comment by paulo27ms** [comment]
Updating to say that the bug that I mentioned was present in the last build is indeed as I say, every time you refresh the tracker data (pulled down or used the refresh tracking metadata button), it'd send information back to MAL and updated the entry even if you didn't change anything.
This is no longer happening in this week's build, it only happens when you track a manga for the first time.

**Comment by arkon** [comment]
Is this still an issue after MAL was rewritten in 0.10.8?

**Comment by paulo27ms** [comment]
Indeed, when you add tracking to a manga, Tachiyomi is sending some information back to MAL for some reason still.
Hopefully this makes it clear:
https://streamable.com/d8ygjy

**Comment by NitroTears** [comment]
I'm happy to work on this! I wanted to clarify some logic I found though. If the user has read some chapters in the app and doesn't have the rereading or completed status on MAL, it will send information to MAL to mark the manga as currently reading. This logic should still say right? If no changes are detected (such as the user wants to start tracking before they start reading), only then would we **not** send any info back to cause an uneccessary update, right?

**Comment by arkon** [comment]
Yes, it should still add the entry to the list if it doesn't already exist. It should only avoid sending an update if all of the data is considered the same.

This notably has to be done in all of the trackers separately (although I'm aware that this level of duplication across the trackers' implementations isn't ideal).

**Comment by NitroTears** [comment]
Would you say that this logic being added into all of the trackers is within the scope of this issue? Or should I just do MAL for now?

**Comment by arkon** [comment]
Should be all of them, although we can do them one by one and leave this issue open.

**Comment by NitroTears** [comment]
Alright cool, I'll try to see if I can avoid having to duplicate the code across the trackers. Worst comes to worst at least the functionality will be in.
